### PR TITLE
Profile payload refactor

### PR DESCRIPTION
**Pull request by phmccarty**
Move the duplicated code into a new function, and have all of the patterns use it if they support it. Also, add a stub for the `make` pattern.

### CODE REVIEW

File: autospec/specfiles.py [hunk 1]
```diff
@@ -572,6 +573,55 @@ def write_license_files(self):
                 file2 = file.replace("/", "_")
                 self._write_strip("cp " + file + " %{buildroot}/usr/share/package-licenses/" + self.name + "/" + file2 + "\n")
 
+    def write_profile_payload(self, pattern=None):
+        """Write the profile_payload specified for this package."""
+        if not pattern:
+            return
+        if not config.profile_payload:
+            return
+        if config.profile_payload and not config.profile_payload[0]:
```
Comment (bryteise) [inline autospec/specfiles.py]: extra check on config.profile_payload

File: autospec/specfiles.py [hunk 2]
```diff
@@ -572,6 +573,55 @@ def write_license_files(self):
...
+                   config.extra_configure,
+                   config.extra_configure64)
+        elif pattern == "cmake":
+            use_subdir = False
+            init = "{0}".format(self.get_profile_generate_flags())
+            post = "{0}".format(self.get_profile_use_flags())
+        else:
+            print_warning(f"profile_payload not supported for {pattern} pattern")
+            return
+        if config.profile_payload and config.profile_payload[0]:
```
Comment (bryteise) [inline autospec/specfiles.py]: extra check on config.profile_payload

File: autospec/specfiles.py [hunk 3]
```diff
@@ -572,6 +573,55 @@ def write_license_files(self):
...
+            if use_subdir and self.subdir:
+                self._write_strip("pushd " + self.subdir)
+            self._write_strip(init)
```
Comment (bryteise) [inline autospec/specfiles.py]: since you check for post, maybe check for init too.

File: autospec/specfiles.py [hunk 4]
```diff
@@ -572,6 +573,55 @@ def write_license_files(self):
...
+                   self.get_profile_generate_flags(),
+                   self.disable_static,
+                   config.extra_configure,
+                   config.extra_configure64)
+        elif pattern == "cmake":
+            use_subdir = False
+            init = "{0}".format(self.get_profile_generate_flags())
+            post = "{0}".format(self.get_profile_use_flags())
+        else:
+            print_warning(f"profile_payload not supported for {pattern} pattern")
```
Comment (bryteise) [inline autospec/specfiles.py]: maybe this should be a hard error for testing

**Comment by bryteise** [comment]
Seems like a reasonable approach.

File: autospec/specfiles.py (hunk 1 above)
Comment (phmccarty) [reply autospec/specfiles.py]: fixed

File: autospec/specfiles.py (hunk 2 above)
Comment (phmccarty) [reply autospec/specfiles.py]: fixed

File: autospec/specfiles.py (hunk 4 above)
Comment (phmccarty) [reply autospec/specfiles.py]: I removed this block in v2

**Comment by phmccarty** [comment]
Changes since the v1 PR:
- Do not special-case the `make` pattern. Instead, if `pattern` is not passed to `write_profile_payload()` (or is `None`), the default behavior is assumed and no special init/post steps are needed.
- Switch to concatenated formatted string literals instead of calling `format`

File: autospec/specfiles.py (hunk 3 above)
Comment (phmccarty) [reply autospec/specfiles.py]: fixed in v2

File: autospec/specfiles.py [hunk 5]
```diff
@@ -572,6 +572,50 @@ def write_license_files(self):
                 file2 = file.replace("/", "_")
                 self._write_strip("cp " + file + " %{buildroot}/usr/share/package-licenses/" + self.name + "/" + file2 + "\n")
 
+    def write_profile_payload(self, pattern=None):
+        """Write the profile_payload specified for this package."""
+        if not config.profile_payload:
+            return
+        if not config.profile_payload[0]:
+            return
```
Comment (bryteise) [inline autospec/specfiles.py]: if not config.profile_payload or not config.profile_payload[0] probably but up to you.

Comment (phmccarty) [reply autospec/specfiles.py]: I will remove the second conditional, since we determined that it's not needed.

**Pull request merged**